scalyr_agent/log_processing.py

```diff
@@ -670,8 +763,91 @@ def readline(self, current_time=None):
                     limit_key=("bad-json-%s" % self.__path),
                 )
 
+        raw_line_length = self.__buffer.tell() - original_buffer_position
+
+        # See if we should be using an extended log line to represent this line.  If we do, we will
+        # essentially iterate over the contents, returning fragments sized to the maximum line size.
+        if raw_line_length > self.__max_line_length:
+            self.__extended_line_buffer = LogFileIterator.ExtendedLineBuffer(
+                result, self.__position, raw_line_length
+            )
+            # As part of the invariant for iterating over the extended log line, we always leave the
+            # buffer position just at the start of the extended log line.
+            self.__buffer.seek(original_buffer_position)
+
+            # If we have a previous fragment position but just reread the extended line (which may happen during
+            # seeks or restoring from a checkpoint), check to make sure we that looks like it is the same one as
+            # we got last time we parsed the line. Otherwise, invalidate the fragment position.  This will cause us to
+            # just copy the extended line again, It may be different if they changed configuration parameters for the
```

Comment:
If we change the config, don't all existing log iterators get closed down and new ones created? In which case we should never see them change?

I think it's still worth checking for this, but it should never happen because of a config change.

Reply:
I updated the comments here to be more explicit.  This can happen when we restore from a checkpoint due to either a agent soft restart or a full restart.  The previous LogFileIterators were shut down, but they still persisted a checkpoint reflecting how they read the line (which is based on the
value of max_line_length and the max_internal_parse_line values).